packages/app/src/containers/Canvas.tsx

```diff
@@ -287,6 +290,11 @@ export default memo(function Canvas() {
                 : 'replace',
             );
             dispatch('interaction', ['maybeMovePoint', point]);
+            const canClose = canClosePath(state, {
+              type: 'point',
+              value: selectedPoint,
+            });
+            if (canClose) dispatch('setIsClosed', true);
```

Comment:
We should also check that shift isn't held, since that should add a point to the selection without closing the path

Reply:
Currently, the pointer will still change to 'no-drop' even if shift is true (but it will not close the path). Should we have another interaction state to handle this use case?

Reply:
Good idea! I added a `keyModifiers` to `ApplicationState` to solve this issue.